src/chrome/chromeDebugAdapter.ts

```diff
@@ -472,6 +481,60 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
         } else if (notification.hitBreakpoints && notification.hitBreakpoints.length) {
             reason = 'breakpoint';
 
+            const hitBreakpoints = notification.hitBreakpoints;
+            let allStopOnEntryBreakpoints = true;
+
+            // If there is a breakpoint which is not a stopOnEntry breakpoint, we appear as if we hit that one
+            // This is particularly done for cases when we end up with a user breakpoint and a stopOnEntry breakpoint on the same line
+            hitBreakpoints.forEach(bp => {
+                if (!this._stopOnEntryBreakpointIdToRequestedFileName.has(bp)) {
+                    notification.hitBreakpoints = [bp];
+                    allStopOnEntryBreakpoints = false;
+                }
+            });
+
+            // If all the breakpoints on this point are stopOnEntry breakpoints
+            // This will be true in cases where it's a single breakpoint and it's a stopOnEntry breakpoint
+            // This can also be true when we have multiple breakpoints and all of them are stopOnEntry breakpoints, for example in cases like index.js and index.bin.js
+
+            if (allStopOnEntryBreakpoints) {
+                // Get the mapped url of the file where execution is paused
+                const pausedScriptId = notification.callFrames[0].location.scriptId;
+                const pausedScriptUrl = this._scriptsById.get(pausedScriptId).url;
+
+                const mappedUrl = this._pathTransformer.scriptParsed(pausedScriptUrl);
+                let normalizedMappedUrl: string;
+                if (mappedUrl !== undefined) {
+                    normalizedMappedUrl = path.normalize(mappedUrl);
+                }
+
+                // If the file has unbound breakpoints, make sure to resolve them first and then decide to continue or not
+                if (this._pendingBreakpointsByUrl.has(normalizedMappedUrl)) {
+                    this.resolvePendingBreakpoint(this._pendingBreakpointsByUrl.get(normalizedMappedUrl))
```

Comment:
I'd change this method to return some information about the breakpoint, so you can deduce _userBreakpointOnLine1Col1 from there, rather than having to store it on an instance variable...

Reply:
I don't wanna change the function signature just for my use case since I think it makes more sense the way it is right now. Do you see any major issue with the use of an instance variable?

Reply:
Yes I need this per file that's why I reset it every time if it becomes true. Since I am waiting for the resolvePendingBP request to complete before resetting it, I figured this should be okay and we won't need to store it per file since the operation will be synchronous for different files.

Reply:
So it relies on the scriptParsed and onPause happening right next to each other, and scriptParsed coming first? That's probably a safe assumption.

Reply:
Would help to document these assumptions.